Fix DeprecationWarning on Python3: use html.escape

brutasse opened the pull request:
Hello,

Python 3 raises a DeprecationWarning when using `cgi.escape`. `html.escape` is the replacement. This PR avoids the warning with a fallback to `cgi.escape` on Python 2.

Thanks!

theSage21 commented:
This might go into compat for compatibility. Am I right @Alir3z4 ? Also, @brutasse I cannot think of any reason for the Apostrophy test to fail. Any Ideas?

Alir3z4 commented on html2text/__init__.py:
```diff
@@ -4,7 +4,11 @@
 import sys
-import cgi
+
+try:
```
This need to be in `compat.py` module.

Alir3z4 commented:
Tests for Python 3 are falling I think due to some Unciode/Encoding issues from `escape` method.

brutasse commented:
@Alir3z4 I moved the imports to the `compat` module and fixed the Python 3 tests. Commit summary has all the information: https://github.com/Alir3z4/html2text/pull/126/commits/b2b2fb28092c74167816e407067958486f102a9c

Let me know if anything else is needed.

Alir3z4 commented:
@brutasse thanks for the patch and applying further changes.
@theSage21 Thanks for merging this, could you please update the ChangeLog as well?
